### ISSUE TITLE
help with cell annotations

### ISSUE THREAD

**Issue by Gibbsdavidl** [maximilianh/cellBrowser#156]
Hi there,

Loving the package!
I'm trying to add an additional column to the cell metadata. But, when I want to color cells by cell type ... it keeps complaining that the field ('cell_id' below) is a unique identifier ... would really appreciate some help!   

**The meta.tsv file looks like:**
        batch   UMI Count       Percent Mitochond.      pheno   sample  Expressed Genes leiden  cell_id
AAAGTCCCAAGTGGGT-0      0       18572   0.015507215     T       genecounts/E02C 1616    9       B_cell_IgA_secreting
AAAGTCCTCGGTCTGG-0      0       6601    0.052113317     T       genecounts/E02C 1718    9       B_cell_IgA_secreting
AACAAGACACATCATG-0      0       25416   0.011724898     T       genecounts/E02C 2266    9       B_cell_IgA_secreting
AACACACAGTCATACC-0      0       31278   0.011509688     T       genecounts/E02C 2319    9       B_cell_IgA_secreting
AACAGGGTCCTACGAA-0      0       6433    0.05394062      T       genecounts/E02C 1445    9       B_cell_IgA_secreting

**and the conf file looks like:**
name='E02'
shortLabel='E02'
exprMatrix='exprMatrix.tsv.gz'
meta='meta.tsv'
geneIdType='auto'
defColorField='leiden'
labelField='leiden'
enumFields=['leiden','cell_id']
coords=[
    {
        "file": "umap_coords.tsv",
        "shortLabel": "UMAP"
    }
]
markers = [{"file": "markers.tsv", "shortLabel":"Cluster Markers"}]
quickGenesFile="quickGenes.tsv"

Any idea?  THANK YOU!!

**Comment by maximilianh** [comment]
Thanks! Glad that someone likes it.

are you able to share the full meta.tsv file? I'd love to look at it, I
suspect some newline or tab related problem or something strange with the
header lines...

**Comment by Gibbsdavidl** [comment]
Ah! With your hint, I fixed it.

Turned out that I was missing a column name over the cell barcode 'index'.  I guess all columns need a header? Unlike R data.frames where the first column, with no header, can be interpreted as row names.

so it's like:

index  batch    UMI Count   Percent Mitochond.   pheno   sample   Expressed Genes   leiden   cell_id
AAAGTCCCAAGTGGGT-0 0 18572 0.015507215 T genecounts/E02C 1616 9 B_cell_IgA_secreting
AAAGTCCTCGGTCTGG-0 0 6601 0.052113317 T genecounts/E02C 1718 9 B_cell_IgA_secreting

And now I can color by cell types and they come up in the panel on the right.
